src/Workspaces/Core/Portable/Workspace/ProjectSystem/IFileChangeWatcher.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.ProjectSystem
+{
+    internal interface IFileChangeWatcher
+    {
+        IFileChangeContext CreateContext(params WatchedDirectory[] watchedDirectories);
```

Comment:
consider just an IMmutableArray.

Reply:
I did, then realized we had a bunch of places taking advantage of the params so reverted it. Consider this ugliness as a way to encourage the LDM to finish the params improvements. :smile: